### PR TITLE
Add security_options to rmw_create_node argument list

### PR DESCRIPTION
**Pull request by mikaelarguedas**
connects to ros2/ros2#339

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Shouldn't this return an error code instead?

**Comment by mikaelarguedas** [comment]
You're absolutely right, I'll return a runtime_error("Not implemented") here.

**Comment by dirk-thomas** [comment]
This is a C API so the caller can't catch it in C. Therefore I think returning an error codec is more appropriate.

**Comment by mikaelarguedas** [comment]
Yeah good point, returning an error code in a function that usually return a node structure is not convenient though. I'll return NULL and set an RMW error message then. Thanks for pointing it out

File: rmw_opensplice_cpp/src/rmw_node.cpp [hunk 1]
```diff
@@ -38,6 +40,10 @@ rmw_create_node(const char * name, const char * namespace_, size_t domain_id)
     RMW_SET_ERROR_MSG("namespace_ is null");
     return nullptr;
   }
+  if (security_options->enforce_security) {
```
Comment (dirk-thomas) [inline rmw_opensplice_cpp/src/rmw_node.cpp]: You must check that `security_options` is not null before accessing its members.

**Pull request merged**